### PR TITLE
[concept entry] Dart map putIfAbsent function

### PR DESCRIPTION
**Pull request by sambunyan**
Sam_p_Bunyan

### Description
-  I have added the document putifabsent.md in docs/content/dart/concepts/map/terms//putifabsent/putifabsent.md.

### Issue Solved
- Closes issue number #4373 
 
### Type of Change

- Adding a new entry

### Checklist

- [X ] All writings are my own.
- [X ] My entry follows the Codecademy Docs style guide.
- [ X] My changes generate no new warnings.
- [X ] I have performed a self-review of my own writing and code.
- [X ] I have checked my entry and corrected any misspellings.
- [X ] I have made corresponding changes to the documentation if needed.
- [X ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Review by helfese — APPROVED** [review]

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @sambunyan, you have committed two files and only `content/dart/concepts/map/terms/putifabsent/putifabsent.md` this is valid. Please delete the other file so that I can then review it and suggest the changes.

**Comment by sambunyan** [comment]
hello @avdhoottt, I have deleted the extra depository and the files within from my merging branch, only content/dart/concepts/map/terms/putifabsent/putifabsent.md is present now.

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+Title: '.putifabsent()'
+Description: '.putifabsent() is used to insert a key-value pair into a map if the key is not already present. '
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:3]:
```suggestion
Description: 'Inserts a key-value pair into a map if the key is not already present.'
```

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+---
+Title: '.putifabsent()'
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:2]:
```suggestion
Title: '.putIfAbsent()'
```

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Dart'
+  - 'map'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The .putifabsent() method in Dart is used to insert a key-value pair into a map if the key is not already present. If the key is already present, it does nothing and returns the existing value associated with the key.
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:15]:
```suggestion
The **`.putifabsent()`** method in Dart is used to insert a key-value pair into a map if the key is not already present. If the key is already present, it does nothing and returns the existing value associated with the key.
```

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+## Syntax
+
+```pseudo
+V putIfAbsent(K key, V ifAbsent())
+```
+
+- `Key`: The key to check for or add in the map.
+- `ifabsent`: A function that returns the value to be associated with the key if it is not already present in the map.
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:24]:
```suggestion
- `key`: The key to check for or add to the map.
- `ifAbsent`: A function that returns the value associated with the key if it is not already in the map.
```

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+void main() {
+  var map = {'key1': 1, 'key2': 2};
+// 'key3' is not present, so the value 3 will be provided by the ifabsent function
+map.putifabsent('key3', () => 3); // {'key1': 1, 'key2': 2, 'key3': 3};
+
+  // 'key1' is present, so the ifabsent function is not invoked
+map.putifabsent('key1', () => 5); // Notice the output stays the same {'key1': 1, 'key2': 2, 'key3': 3};
+
+print(map); // output {'key1': 1, 'key2': 2, 'key3': 3}
+
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:40]:
```suggestion
void main() {
  var map = {'key1': 1, 'key2': 2};
  // 'key3' is not present, so the value 3 will be provided by the ifAbsent function
  map.putIfAbsent('key3', () => 3); // {'key1': 1, 'key2': 2, 'key3': 3}
  // 'key1' is present, so the ifAbsent function is not invoked
  map.putIfAbsent('key1', () => 5); // Notice the output stays the same {'key1': 1, 'key2': 2, 'key3': 3}
  print(map); // output {'key1': 1, 'key2': 2, 'key3': 3}
}
```
The code had a missing bracket and the method must be called by its exact name as it is case-sensitive.

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+map.putifabsent('key1', () => 5); // Notice the output stays the same {'key1': 1, 'key2': 2, 'key3': 3};
+
+print(map); // output {'key1': 1, 'key2': 2, 'key3': 3}
+ 
+```
+
+Here is the output for the above code:
+
+```
+{'key1': 1, 'key2': 2, 'key3': 3}
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:46]:
```suggestion
{key1: 1, key2: 2, key3: 3}
```

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  // 'key1' is present, so the ifabsent function is not invoked
+map.putifabsent('key1', () => 5); // Notice the output stays the same {'key1': 1, 'key2': 2, 'key3': 3};
+
+print(map); // output {'key1': 1, 'key2': 2, 'key3': 3}
+ 
+```
+
+Here is the output for the above code:
+
+```
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:45]:
```suggestion
```shell
```
The output block should be inside ` ```shell`.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey Author, I have suggested some changes, can you make those?

**Comment by sambunyan** [comment]
hello @avdhoottt I have made the suggested changes.

**Review by avdhoottt — APPROVED** [review]
PR is ready for the second review.

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+```pseudo
+V putIfAbsent(K key, V ifAbsent())
+```
+
+- `key`: The key to check for or add to the map.
+- `ifAbsent`: A function that returns the value associated with the key if it is not already in the map.
+
+## Example
+
+In this example, the key 'key3' is not present in the map, so the ifAbsent function () => 3 is invoked to provide the value 3. The key 'key1' and 'key2' are present so the ifAbsent function is not invoked and the existing values are returned.
```
Comment (ishg-153) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:28]:
```suggestion
In this example, the key `key3` is not present in the map, so the `ifAbsent()` function `() => 3` is invoked to provide the value 3. The keys `key1` and `key2` are present so the `ifAbsent()` function is not invoked and the existing values are returned.
```

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+```dart
+void main() {
+  var map = {'key1': 1, 'key2': 2};
+  // 'key3' is not present, so the value 3 will be provided by the ifAbsent function
+  map.putIfAbsent('key3', () => 3); // {'key1': 1, 'key2': 2, 'key3': 3}
+  // 'key1' is present, so the ifAbsent function is not invoked
+  map.putIfAbsent('key1', () => 5); // Notice the output stays the same {'key1': 1, 'key2': 2, 'key3': 3}
+  print(map); // output {'key1': 1, 'key2': 2, 'key3': 3}
+}
```
Comment (ishg-153) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:38]:
```suggestion
void main() {
  var map = {'key1': 1, 'key2': 2};
  // 'key3' is not present, so the value 3 will be provided by the ifAbsent function
  map.putIfAbsent('key3', () => 3); // {'key1': 1, 'key2': 2, 'key3': 3}
  // 'key1' is present, so the ifAbsent function is not invoked
  print(map.putIfAbsent('key1', () => 5)); // Notice the output is the existing value associated with key1
  print(map); // output {'key1': 1, 'key2': 2, 'key3': 3}
}
```

It makes sense to show how the method returns the existing value of the key.

File: content/dart/concepts/map/terms/putifabsent/putifabsent.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+Here is the output for the above code:
+
+```shell
+{key1: 1, key2: 2, key3: 3}
```
Comment (ishg-153) [inline content/dart/concepts/map/terms/putifabsent/putifabsent.md:44]:
```suggestion
1
{key1: 1, key2: 2, key3: 3}
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@sambunyan I have suggested some changes. Please make those and ping me once done.

**Comment by sambunyan** [comment]
@ishg-153 I have applied the suggested changes

**Review by ishg-153 — APPROVED** [review]
@sambunyan Great work! Ready to merge.

**Pull request merged**